supervisor/updater.py

```diff
@@ -242,11 +242,17 @@ async def fetch_data(self):
 
             # Update HassOS version
             if self.sys_os.board:
-                events.append("os")
-                self._data[ATTR_HASSOS] = AwesomeVersion(
-                    data["hassos"][self.sys_os.board]
-                )
                 self._data[ATTR_OTA] = data["ota"]
+                if self.sys_os.board in data["hassos"]:
+                    events.append("os")
+                    self._data[ATTR_HASSOS] = AwesomeVersion(
+                        data["hassos"][self.sys_os.board]
+                    )
+                else:
+                    _LOGGER.warning(
```

Comment:
Should we here mark it as unsupported?

Reply:
Hm, just because we don't support it in OS 7, I think OS 6.6 should remain supported... I'd say we only mark it as unsupported once we consider that particular OS version as unsupported.

Reply:
But they will never be able to update?
Shouldn't we note that as soon as possible so they can find something else to run it on?

Reply:
Hence that message as a warning? It just feels to me that marking it as unsupported already today seems somewhat harsh. What implications does it have?

Reply:
That warning will be missed by most.
We never really supported the missing boards.

The implications will be a note in the UI and logs

Reply:
How about marking it as unsupported as soon as its removed from the version file but leave the boards in the version file pointing to 6.6 for a while? This way people using <6.6 still get the update to the last supported version, and we have a bit of a transition time.

Reply:
There is one problem though with that approach, the GH action updates versions for all boards unconditionally. So that would need some fixes there :sweat:

Reply:
That would be a better approach indeed 👍
But that's out of the scope in this PR, I'd say is fine as is, and we make it better in a follow up when we have adjusted that logic